libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/preferences/PreferenceText.kt

```diff
@@ -108,35 +96,35 @@ fun PreferenceText(
                     color = tintColor ?: enabled.toSecondaryEnabledColor(),
                 )
             }
-        }
-        if (currentValue != null) {
-            Text(
-                modifier = Modifier
-                    .align(Alignment.CenterVertically)
-                    .padding(start = 16.dp, end = 8.dp),
-                text = currentValue,
-                style = ElementTheme.typography.fontBodyXsMedium,
-                color = enabled.toSecondaryEnabledColor(),
-            )
-        } else if (loadingCurrentValue) {
-            CircularProgressIndicator(
-                modifier = Modifier
-                    .progressSemantics()
-                    .padding(start = 16.dp, end = 8.dp)
-                    .size(20.dp)
-                    .align(Alignment.CenterVertically),
-                strokeWidth = 2.dp
-            )
-        }
-        if (showEndBadge) {
-            val endBadgeStartPadding = if (currentValue != null || loadingCurrentValue) 8.dp else 16.dp
-            RedIndicatorAtom(
-                modifier = Modifier
-                    .align(Alignment.CenterVertically)
-                    .padding(start = endBadgeStartPadding)
-            )
-        }
-    }
+        },
+        trailingContent = ListItemContent.Custom {
```

Comment:
Same about nullability here. If there's nothing to display it should be `null`, not an empty row.